modules/drivers/oracle/src/metabase/driver/oracle.clj

```diff
@@ -317,6 +319,53 @@
   [driver _ field-or-value]
   (sql.qp/unix-timestamp->honeysql driver :seconds (hx// field-or-value (hsql/raw 1000000))))
 
+(defn- time-zoned-trunc
+  "Same as [[trunc]], but truncates `x` to `unit` in the results timezone
+   if `x` is a timestamp with time zone."
+  [unit x]
+  (let [x (cond-> x
+             (hx/is-of-type? x #"(?i)timestamp(\(\d\))? with time zone")
```

Comment:
I think you've got logical grouping parens `\( ... \)` in addition to match group parens `( ... )` here. I think this could be changed to just `#"(?i)timestamp\(\d+\)? with time zone"`.
I added `\d+` just in case a db has a timestamp(10) or anything silly. Also, this regex is matched with `re-find` so an optional character at the beginning can be ignored.

Reply:
Good catch, I copied this from the `database-type->base-type` function and the convert-timezone implementation. Might be worth updating those too. Though oracle's max precision is 9, I think we're safe without the `+` https://github.com/metabase/metabase/blob/5e22f2e0726bde2a91995b553007d5918a17031b/modules/drivers/oracle/src/metabase/driver/oracle.clj#L43

Reply:
Wait, actually your suggestion doesn't work. 

`#"(?i)timestamp\(\d+\)? with time zone"` doesn't match `timestamp with time zone`, which has a default precision of 6.